### ISSUE TITLE
Create Auxiliary Variable Storage

### ISSUE THREAD

**Issue by JaredCrean2** [OptimalDesignLab/PDESolver.jl#3]
Precomputing and storing extra variables is going to be helpful, so we need a way to store and access them efficiently.  @jehicken has suggested using macros to access the array so the user doesn't need to remember which variables are stored where.  A basic implimentation is in a55d9cc.

**Comment by JaredCrean2** [comment]
The benefits of using macros rather than functions to access the variables are unclear, especially if the locations of the variables in the array are statically known.  I'm not sure the complexity (and difficulty of debugging) macros is worth it.

On the other hand, if we want to store (or not store) different variables based on runtime options, using macros would be great because the location of a variable could  be looked up from a dictionary at compile time in the macro, but used at runtime without the performance hit of a dictionary lookup.

**Comment by jehicken** [comment]
Here is an updated example of how we might use macros to set variables.  This is applied to the conservative variables --- which, in practice, we would likely just access using indices --- but it shows how the idea can be applied.

```julia
function EulerFlux(u::AbstractArray{Float64,1},                                        
                   flux::AbstractArray{Float64,1})                                     
  ρ = @ρ(u)                                                                            
  ρu = @ρu(u)                                                                          
  E = @Energy(u)                                                                       
  press = (γ-1.0)*(E - 0.5*ρu*ρu/ρ)                                                    
  flux[1] = ρu                                                                         
  flux[2] = ρu*ρu/ρ + press                                                            
  flux[3] = E*ρu/ρ                                                                     
end 
```
where the macros are defined as
```julia
macro ρ(u)   
  return :($u[1])     
end                        
macro ρu(u)       
  return :($u[2]) 
end
macro Energy(u) 
  return :($u[3]) 
end 
```

Running this against the version of the code where the variables are accessed directly, I get approximately the same time.

**Comment by JaredCrean2** [comment]
Apparently I forgot to include the file with the macros in the earlier commit.  The are in ae9e66d, which I think is in line with your example here.

**Comment by JaredCrean2** [comment]
This appears to be in good shape for now.  As auxiliary variables are used more we can refine them.  closed by  85f8556
